<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="gui" default="true"/>
    <arg name="paused" default="false"/>
    <arg name="repair_mission" default="false"/>
    <arg name="model_path" default=""/>

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="worlds/mangalia.world"/>
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="$(arg gui)"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
        <arg name="verbose" value="true" />
    </include>

    <group if="$(arg repair_mission)">
        <arg name="uav_name" default="hector" />
        <arg name="sdf_uav_file" default="$(arg model_path)/MIMRee/mimree_simulator/models/mimree_uav/model.sdf" />

        <param name="$(arg uav_name)_description" textfile="$(find mimree_description)/urdf/$(arg uav_name).urdf" />

        <node name="$(arg uav_name)_spawn" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen" args="-file $(arg sdf_uav_file) -sdf -x -0.0 -y -0.0 -z 7.42 -R 0. -P 0. -Y 0. -model $(arg uav_name)"/>
        <!-- args x y z yaw pitch roll frame_id child_id time_in_ms -->
        <!-- <node name="$(arg uav_name)_scanningLidar_broadcaster" pkg="tf" type="static_transform_publisher" args="0.191 0 0.591936 -1.5707 1.5707 0 $(arg uav_name) $(arg uav_name)_scanningLidar 33" />
        <node name="$(arg uav_name)_altLidar_broadcaster" pkg="tf" type="static_transform_publisher" args="0.151 0 0.591936 0 1.5707 0 $(arg uav_name) $(arg uav_name)_altLidar 33" /> -->
        <!-- <node name="$(arg uav_name)_tf_broadcaster" pkg="mimree_description" type="mimree_tf_broadcaster.py" args="$(arg uav_name) mimree_uav" /> -->
        <node name="$(arg uav_name)_altLidar_tf_broadcaster" pkg="mimree_description" type="mimree_tf_broadcaster" args="$(arg uav_name) mimree_uav altLidar" />
        <node name="$(arg uav_name)_scanningLidar_tf_broadcaster" pkg="mimree_description" type="mimree_tf_broadcaster" args="$(arg uav_name) mimree_uav scanningLidar" />
        <node name="$(arg uav_name)_altLidar_to_cloud" pkg="mimree_description" type="laser_to_pc" args="$(arg uav_name) altLidar" />
        <node name="$(arg uav_name)_scanningLidar_to_cloud" pkg="mimree_description" type="laser_to_pc" args="$(arg uav_name) scanningLidar" />
    </group>

    <group unless="$(arg repair_mission)">
        <arg name="uav_name" default="hector" />
        <arg name="asv_name" default="halcyon" />
        <arg name="sdf_uav_file" default="$(arg model_path)/MIMRee/mimree_simulator/models/mimree_uav/model.sdf" />
        <arg name="sdf_asv_file" default="$(arg model_path)/MIMRee/mimree_simulator/models/mimree_asv/model.sdf" />

        <param name="$(arg uav_name)_description" textfile="$(find mimree_description)/urdf/$(arg uav_name).urdf" />
        <param name="$(arg asv_name)_description" textfile="$(find mimree_description)/urdf/$(arg asv_name).urdf" />

        <node name="$(arg uav_name)_spawn" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen" args="-file $(arg sdf_uav_file) -sdf -x -14.0 -y -19.65 -z 2.0 -R 0. -P 0. -Y 0. -model $(arg uav_name)"/>
        <node name="$(arg uav_name)_altLidar_tf_broadcaster" pkg="mimree_description" type="mimree_tf_broadcaster" args="$(arg uav_name) mimree_uav altLidar" />
        <node name="$(arg uav_name)_scanningLidar_tf_broadcaster" pkg="mimree_description" type="mimree_tf_broadcaster" args="$(arg uav_name) mimree_uav scanningLidar" />
        <node name="$(arg uav_name)_altLidar_to_cloud" pkg="mimree_description" type="laser_to_pc" args="$(arg uav_name) altLidar" />
        <node name="$(arg uav_name)_scanningLidar_to_cloud" pkg="mimree_description" type="laser_to_pc" args="$(arg uav_name) scanningLidar" />

        <node name="$(arg asv_name)_spawn" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen" args="-file $(arg sdf_asv_file) -sdf -x -10.0 -y -20.0 -z 0.0 -R 0. -P 0. -Y 0. -model $(arg asv_name)"/>
        <node name="$(arg asv_name)_altLidar_tf_broadcaster" pkg="mimree_description" type="mimree_tf_broadcaster" args="$(arg asv_name) mimree_asv altLidar" />
        <node name="$(arg asv_name)_altLidar_to_cloud" pkg="mimree_description" type="laser_to_pc" args="$(arg asv_name) altLidar" />
    </group>

</launch>
